src/mpid/ch4/netmod/ucx/ucx_progress.c

```diff
@@ -78,44 +76,31 @@ static int am_handler(void *msg, size_t msg_sz)
     if (target_cmpl_cb) {
         target_cmpl_cb(rreq);
     }
-
-  fn_exit:
-    return mpi_errno;
-}
-
-static void handle_am_recv(void *request, ucs_status_t status, ucp_tag_recv_info_t * info)
-{
-    return;
 }
 
 int MPIDI_UCX_progress(int vci, int blocking)
 {
     int mpi_errno = MPI_SUCCESS;
     ucp_tag_recv_info_t info;
     MPIDI_UCX_ucp_request_t *ucp_request;
-    void *am_buf;
     ucp_tag_message_h message_handle;
+
     /* check for active messages */
-    message_handle =
-        ucp_tag_probe_nb(MPIDI_UCX_global.worker, MPIDI_UCX_AM_TAG, MPIDI_UCX_AM_TAG, 1, &info);
-    while (message_handle) {
+    while ((message_handle =
+            ucp_tag_probe_nb(MPIDI_UCX_global.worker, MPIDI_UCX_AM_TAG, MPIDI_UCX_AM_TAG, 1,
+                             &info))) {
```

Comment:
One of my personal preference is the following pattern:
```
while (1) {
    type var = func(...);
    if (var) {
        break;
    }
    ... body ...
}
```

It avoids long, nested condition expression whose main purpose is for side effects.

Reply:
Works for me. I don't love the assignment-in-conditional as it is. Will change.